tests/test_control_flow.py

```diff
@@ -0,0 +1,134 @@
+from unittest.mock import patch
+
+import torch
+
+import torchdynamo.testing
+from torchdynamo.logic.control_flow import cond
+
+
+class ControlFlowTests(torchdynamo.testing.TestCase):
+    @patch.object(torchdynamo.config, "fake_tensor_propagation", False)
+    def test_simple_condition(self):
+        from_true = None
+        from_false = None
+
+        @torchdynamo.optimize("eager", nopython=True)
+        def foo():
+            nonlocal from_true
+            nonlocal from_false
+
+            def compute(pred, z):
+                def when_true(x):
+                    return x + x
+
+                def when_false(x):
```

Comment:
Yeah - I was considering this. I kind of like the idea of forcing when_true and when_false to have identical contracts w/r/t to input and return. No strong opinions, and it would not be overly onerous to add at all - do you have a preference?

Reply:
We should understand why exir was designed that way.  In some cases (e.g. `if len(x) % 2 == 0`) a single input can't go down both sides of the control flow.

Reply:
That's a great point. I can extend it to cover both T and F flows.

Reply:
Added